### ISSUE TITLE
Replace all usages of xmlstarlet with getCheckstylePomVersion from util.sh

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#12726]
There are a few usages of 
```sh
xmlstarlet sel -N pom=http://maven.apache.org/POM/4.0.0 \
                             -t -m pom:project -v pom:version pom.xml | sed "s/-SNAPSHOT//"
```
that can be replaced with `getCheckstylePomVersion` from `./.ci/util.sh`.

![image](https://user-images.githubusercontent.com/23459549/216835589-f022de65-f9c8-4542-97ee-7d32fff6749a.png)

Even better, we can make new util function like:
```bash
function getCheckstylePomVersionWithoutSnapshot {
  getCheckstylePomVersion | sed "s/-SNAPSHOT//"
}
```

**Comment by stoyanK7** [comment]
On it.

**Comment by romani** [comment]
Please be accurate
https://github.com/checkstyle/checkstyle/blob/dac1a2f07f594e9abe36d84f330d8e097a9a1cdc/.ci/util.sh#L17

Use whole maven execution, it is heavy operation. It will trigger whole maven jar download.

If build job is not using maven, better to not reuse such util

**Comment by rnveach** [comment]
Maybe we should add a note and verify if we do use it in non-maven places. The PR that prompted this issue might need to be relooked at since I thought we should use this there.

**Comment by stoyanK7** [comment]
All files who use `xmlstarlet`(to extract version) except `close-create-milestone.sh` invoke `mvn` at a later stage in the script.

@rnveach The PR that prompted the issue was `bump-version.sh` and it uses `mvn`: https://github.com/checkstyle/checkstyle/pull/12722#pullrequestreview-1284251241

**Comment by romani** [comment]
Yes

**Comment by nrmancuso** [comment]
Closed via #12732
